### ISSUE TITLE
Expose an HTTP API for turning a Stripe payment token into a Stripe charge

### ISSUE THREAD

**Issue by exarkun** [PrivateStorageio/PaymentServer#24]
Stripe.js works by sending payment details to a Stripe server and getting back a token, then sending the token to the application backend.  The application backend is required to send the token to a Stripe server along with parameters to create a charge against the payment details represented by the token.  Then the application backend can return success or failure to the web page which can show it to the user.

Implement the part of the application backend that takes a token, sends it to stripe to create a charge, and returns a success/failure result to the web page.

**Comment by exarkun** [comment]
This is basically the thing we need to do instead of having the webhook that we already have (and see #21).

I expect this API looks something like:

```
POST /v1/stripe/charge HTTP/...
Content-Type: application/json

{"token": "tok_...", "voucher": "..."}
```

And the response will be OK if Stripe is happy with this and some error code if it is not or something else goes wrong.

The result should be a Charge in Stripe against the given token with the given "voucher" in its metadata (the existing Stripe webhook shows how the metadata is expected to be structured).

And if the Charge succeeds then state should be inserted into the VoucherDatabase marking the voucher as paid.

Idempotency would be really great too.  If that proves overly complicated then it can be kicked off to a follow-up ticket.  Stripe tokens might be single use though so it might not be very complicated.

**Comment by exarkun** [comment]
Fixed by #30
